configure

```diff
@@ -51,13 +51,33 @@ sanitycheck() {
     exit 1
 }
 
+checkoption() {
+    # $1 = arg
+    option="${1#*--}"
+    action="${option%%-*}"
+    name="${option#*-}"
+    if [ ${default_options[$name]+_} ]; then
+        case "$action" in
+            enable) meson_options[$name]=true;;
+            disable) meson_options[$name]=false;;
+            *) echo -e "\e[1;33mINFO\e[0m: Ignoring unknown action '$action'";;
+        esac
+    else
+        echo -e "\e[1;33mINFO\e[0m: Ignoring unknown option '$option'"
+    fi
+}
+
 sanitycheck MESON 'meson'
 sanitycheck MESONTEST 'mesontest'
 sanitycheck NINJA 'ninja' 'ninja-build'
 
-enable_docs='-Denable-gtk-doc=false'
-enable_man='-Denable-man=false'
-enable_introspection=''
+declare -A default_options=(
```

Comment:
I like this idea, but not all options take a boolean value, so maybe `checkoption()` should take the argument as well, like:

    *) opt=$1; read_arg arg "$@" || shift; checkoption $opt $arg ;;